.github/workflows/publish-image.yml

```diff
@@ -0,0 +1,63 @@
+name: Publish Docker image
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - ready_for_review
+      - synchronize
+  merge_group:
+  workflow_dispatch:
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  push_to_registry:
+    name: Push Docker image to Docker Hub
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+      attestations: write
+      id-token: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Get Current date
+        run: echo "RELEASE_DATE=$(date +'%y.%m.%d')" >> $GITHUB_ENV
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@9ec57ed1fcdbf14dcef7dfbe97b2010124a938b7
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          labels: |
+            org.opencontainers.image.description=GOV.UK One Login Simulator
+            org.opencontainers.image.version=${{ env.RELEASE_DATE }}.${{ github.run_number }}
+          tags: ${{ env.RELEASE_DATE }}.${{ github.run_number }}
```

Comment:
[from the discovery output](https://docs.google.com/document/d/1u-xKH3b19CdA4XTpeKAGM53clWtdHL0kJYmN-uE3dKc/edit#heading=h.tdwiw5dhwh9), we want to generate a git tag as well as a docker tag, and then generate the next tag from the previous one - ie not just the run number